### PR TITLE
Long calls patch, close #6

### LINKED ISSUE #6: Long Calls Dropped
Calls that are longer than the check interval never get returned. The beginning and end of a call both have to be within the check period to get counted. Increasing the check interval is a temporary mitigation that introduces some more latency and isn't flexible.

### PR DESCRIPTION
**Pull request by spdconvos**
A naive attempt on my part

### CODE REVIEW

File: Bot.py [hunk 1]
```diff
@@ -58,22 +63,46 @@ def __init__(self) -> None:
...
+        Args:
+            calls (list): The complete list of calls scraped.
+
+        Returns:
+            list: A filtered list of calls.
+        """
+        res = []
+        for call in calls:
+            # If the call is already in the cache skip.
+            if call["id"] in self.cache.keys():
```
Comment (AetherUnbound) [inline Bot.py]: I _think_ this can just be:
```suggestion
            if call["id"] in self.cache:

```

File: Scraper.py [hunk 2]
```diff
@@ -47,6 +47,13 @@ def _timestamp(self) -> str:
...
         # There's some delay in the API so you need to delay the timestamp a little.
         # With about 20 minutes of hair pulling "why is the api not returning anything!!!!" frustration I found that 45 seconds was the smallest round-ish number that got returns.
-        time = datetime.now(pytz.utc) - timedelta(seconds=45)
+        # We also scrape back 5 minutes to check for long calls.
+        time = datetime.now(pytz.utc) - timedelta(seconds=(self.lookback + 45))
+
+        # If it's the first check try to get nothing
+        if self.firstCheck:
+            time = datetime.now(pytz.utc)
+            self.firstCheck = False
```
Comment (AetherUnbound) [inline Scraper.py:56]: So simple! Very nice :slightly_smiling_face:

File: Bot.py [hunk 3]
```diff
@@ -7,10 +7,12 @@
 import tweepy
 from tweepy.error import TweepError
 
+from cachetools import TTLCache, ttl
```
Comment (AetherUnbound) [inline Bot.py:10]: Is this second import needed? I think it's only used for the keyword arg on L44, which doesn't require an import.

**Review by AetherUnbound — CHANGES_REQUESTED** [review]
This looks great!! We'll need to add `cachetools` to `requirements.txt` so the container can be rebuild with the new dependency.

File: Bot.py (hunk 3 above)
Comment (spdconvos) [reply Bot.py:10]: I think VS Code auto inserted this at some point oops

File: Scraper.py (hunk 2 above)
Comment (spdconvos) [reply Scraper.py:56]: Sometimes quick and dirty works!

**Pull request merged**